app/Console/Commands/Indexer/IncrementalIndexer.php

```diff
@@ -0,0 +1,226 @@
+    {--failed : Run only entries that failed to index last time}';
+
+    protected function promptForMissingArgumentsUsing(): array
+    {
+        return [
+            'mediaType' => ['The media type to index.', 'Valid values: anime, manga']
+        ];
+    }
+
+    private function getExistingIds(string $mediaType): array
+    {
+        $existingIdsHash = "";
+        $existingIdsRaw = "";
+
+        if (Storage::exists("indexer/incremental/$mediaType.json"))
+        {
+            $existingIdsRaw = Storage::get("indexer/incremental/$mediaType.json");
+            $existingIdsHash = sha1($existingIdsRaw);
+        }
+
+        return [$existingIdsHash, $existingIdsRaw];
+    }
+
+    private function getIdsToFetch(string $mediaType): array
+    {
+        $idsToFetch = [];
+        [$existingIdsHash, $existingIdsRaw] = $this->getExistingIds($mediaType);
+
+        if ($this->cancelled)
+        {
+            return [];
+        }
+
+        $newIdsRaw = file_get_contents("https://raw.githubusercontent.com/purarue/mal-id-cache/master/cache/${mediaType}_cache.json");
+        $newIdsHash = sha1($newIdsRaw);
+
+        /** @noinspection PhpConditionAlreadyCheckedInspection */
+        if ($this->cancelled)
+        {
+            return [];
+        }
+
+        if ($newIdsHash !== $existingIdsHash)
+        {
+            $newIds = json_decode($newIdsRaw, true);
+            $existingIds = json_decode($existingIdsRaw, true);
+
+            if (is_null($existingIds) || count($existingIds) === 0)
+            {
+                $idsToFetch = $newIds;
+            }
+            else
+            {
+                foreach (["sfw", "nsfw"] as $t)
+                {
+                    $idsToFetch[$t] = array_diff($existingIds[$t], $newIds[$t]);
+                }
+            }
+
+            Storage::put("indexer/incremental/$mediaType.json.tmp", $newIdsRaw);
+        }
+
+        return $idsToFetch;
+    }
+
+    private function getFailedIdsToFetch(string $mediaType): array
+    {
+        return json_decode(Storage::get("indexer/incremental/{$mediaType}_failed.json"));
+    }
+
+    private function fetchIds(string $mediaType, array $idsToFetch, bool $resume): void
+    {
+        $index = 0;
+        $success = [];
+        $failedIds = [];
+        $idCount = count($idsToFetch);
+        if ($resume && Storage::exists("indexer/incremental/{$mediaType}_resume.save"))
+        {
+            $index = (int)Storage::get("indexer/incremental/{$mediaType}_resume.save");
+            $this->info("Resuming from index: $index");
+        }
+
+        if ($index > 0 && !isset($this->ids[$index]))
+        {
+            $index = 0;
+            $this->warn('Invalid index; set back to 0');
+        }
+
+        Storage::put("indexer/incremental/{$mediaType}_resume.save", 0);
+
+        $this->info("$idCount $mediaType entries available");
+        $ids = array_merge($idsToFetch['sfw'], $idsToFetch['nsfw']);
```

Comment:
Should we sort similarly to indexer:anime|manga?

https://github.com/jikan-me/jikan-rest/blob/bd8562f7907eda6bd85475ae61c3dd75f40b11d5/app/Console/Commands/Indexer/AnimeIndexer.php#L154C1-L155C34

Reply:
Since merging would mean the nsfw are always at the end of the array.

Reply:
We take the diff, and only the diff matters in this case. Not sure if it would make any difference. I'll add it because it looks better on the output. 😸

Reply:
@pushrbx Fair enough. I think I added it to the orignal indexer because it supported the reverse option so it could start from the latest entries.